webcomponents-lite.js

```diff
@@ -48,30 +47,36 @@
     }
   }
 
-  // construct full dependency list
-  var modules = [
-    '../../webcomponents-platform/webcomponents-platform.js',
-    '../../URL/url.js',
-    '../../template/template.js',
-    '../../html-imports/src/html-imports.js',
-    '../../es6-promise/dist/es6-promise.auto.min.js',
-    'pre-polyfill.js',
-    '../../custom-elements/custom-elements.min.js',
-    '../../shadydom/shadydom.min.js',
-    '../../shadycss/shadycss.min.js',
-    'post-polyfill.js',
-    'unresolved.js'
-  ];
-
-  var src = script.getAttribute('src');
-  var path = src.slice(0, src.lastIndexOf(file));
-
-  modules.forEach(function(f) {
-    document.write('<script src="' + path + 'src/' + f + '"></script>');
-  });
-
   // exports
   WebComponents.flags = flags;
 
+  // Feature detect which polyfill needs to be imported.
+  let polyfills = [];
+  if (!('import' in document.createElement('link'))) {
+    polyfills.push('hi');
+  }
+  if (!window.customElements) {
+    polyfills.push('ce');
+  }
+  if (!('attachShadow' in Element.prototype)) {
+    polyfills.push('sd');
+  }
+  if (!('content' in document.createElement('template')) || !window.Promise || !window.URL) {
+    polyfills.push('pf');
+  }
 
+  // TODO(notwaldorf): This is a temporary hack because of pre/post-polyfill.js,
+  // and needs to not exist at all.
```

Comment:
I can't remove it until there's changes in polymer (it also needs to load ShadyCSS right now)